pulp_ansible/app/models.py

```diff
@@ -275,6 +318,381 @@ class Meta:
         unique_together = ("pubkey_fingerprint", "signed_collection")
 
 
+
+class SigstoreSigningService(Content):
+    """
+    A service to generate Sigstore signatures for a given CollectionVersion.
+    Distinct from SigningService objects used to sign artifacts using GPG
+    (does not call an external script provided by the user, but still needs to be registered).
+    Fields:
+        name (models.CharField):
+            Name of the Sigstore signing service.
+        rekor_url (models.TextField):
+            The URL of the Rekor instance to use for logging signatures.
+            Defaults to the Rekor public good instance URL (https://rekor.sigstore.dev).
+        rekor_root_pubkey (models.TextField):
+            A PEM-encoded root public key for Rekor itself.
+        fulcio_url (models.TextField):
+            The URL of the Fulcio instance to use for getting signing certificates.
+            Defaults to the Fulcio public good instance URL (https://fulcio.sigstore.dev).
+        tuf_url (models.TextField):
+            The URL of the TUF metadata repository instance to use.
+            Defaults to the public TUF instance URL
+            (https://sigstore-tuf-root.storage.googleapis.com/).
+        oidc_issuer (models.TextField):
+            The OpenID Connect issuer to use for signing.
+            Defaults to the public OAuth2 server URL (https://oauth2.sigstore.dev/auth).
+        credentials_file_path (models.TextField):
+            Path to the OIDC client ID and client secret file on the server
+            to authentify to Sigstore.
+        ctfe_pubkey (models.TextField):
+            A PEM-encoded public key for the CT log.
+        cert_identity (models.TextField):
+            A unique identity string corresponding to the OIDC identity
+            present as the SAN in the X509 certificate.
+        enable_interactive (models.BooleanField):
+            Enable Sigstore's interactive browser flow.
+            Defaults to False.
+    """
+
+    TYPE = "sigstore_signing_service"
+
+    PUBLIC_OIDC_ISSUERS = (
+        "https://accounts.google.com",
+        "https://oauth2.sigstore.dev/auth",
+        "http://dex-idp:8888/auth",
+    )
+    PUBLIC_REKOR_URL = "https://rekor.sigstore.dev"
+    PUBLIC_FULCIO_URL = "https://fulcio.sigstore.dev"
+    PUBLIC_TUF_URL = "https://sigstore-tuf-root.storage.googleapis.com/"
+    PUBLIC_ISSUER_URL = "https://oauth2.sigstore.dev/auth"
+
+    name = models.CharField(db_index=True, unique=True, max_length=64)
+    rekor_url = models.TextField(default=PUBLIC_REKOR_URL)
+    rekor_root_pubkey = models.TextField(null=True)
+    fulcio_url = models.TextField(default=PUBLIC_FULCIO_URL)
+    tuf_url = models.TextField(default=PUBLIC_TUF_URL)
+    oidc_issuer = models.TextField(default=PUBLIC_ISSUER_URL)
+    credentials_file_path = models.TextField(null=True)
```

Comment:
Important note: this field is just here temporarily to test the signing workflow by retrieving the OIDC credentials from a file stored in plaintext on the server. This is not the approach we will be going for; I will replace it once an alternative flow to get credentials has been validated.